### PR TITLE
Add support for Couchbase for DocStore & IndexStore

### PR DESCRIPTION
**Pull request by nithishr**
# Description

This PR adds support to use Couchbase as the storage backend for Nodes (Docstore) & Index metadata (IndexStore).

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Feature was tested end to end locally.

- [x] I added new unit tests to cover this change
- [ ] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-couchbase/poetry.lock
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-couchbase/poetry.lock:1]: No need to commit lock files -- lets remove the ones added

**Review by logan-markewich — COMMENTED** [review]
Can you add some unit tests? (mocking out the client might be required if couchbase does not have an in-memory mode)

Ideally we get 50% coverage minimum :)

**Comment by nithishr** [comment]
> Can you add some unit tests? (mocking out the client might be required if couchbase does not have an in-memory mode)
> 
> Ideally we get 50% coverage minimum :)

It looks like mocking is the only way. `skipif()` does not look to be helping.

**Comment by logan-markewich** [comment]
skipif is fine, its just not counted to coverage because its not run in CICD

But, assuming you run it locally, better than nothing!

**Pull request merged**